Bulk Site Creator V2, remove dependency on icommons-common and DCCSW

Chris-Thornton-Harvard opened the pull request:
Merging some changes from develop into the feature branch. Removes dependency on DCCSW and icommons-common
- Adds template tags from icommons-common
- Removes icommons-common and DCCSW from requirements and settings
- Update permissions to use lti_school_permissions #363
- Use modules broken out from icommons-commons #363
- Migration cleanup to remove references to old permissions module
- Adds async_operations module #355
- Adds the CanvasSchoolTemplate model to CAAT TLT-4439

Note: since the canvas_school_template table already exists, a fake migrate should be run. This will insert a record into the django_migrations table for the model but not apply any changes
python manage.py migrate  canvas_account_admin_tools --fake

cmurtaugh commented on canvas_account_admin_tools/templatetags/django_helpers.py:
```diff
@@ -0,0 +1,17 @@
+    return getattr(settings, name, "")
+
+
+@register.filter(is_safe=True)
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you have set the `is_safe` flag to `True` for a custom filter. The `is_safe` flag tells Django not to apply escaping on the value returned by the filter even though the input is escaped. This could expose your application to cross-site scripting (XSS) vulnerabilities. We recommend that you remove `is_safe` and use `mark_safe()` to explicitly mark safe content.

[Learn more](https://cwe.mitre.org/data/definitions/79.html)

cmurtaugh reviewed (commented):
Looks good -- the only part I have a question about is why the CanvasSchoolTemplate model is placed under `common` rather than in the tool app.

Chris-Thornton-Harvard commented:
I originally placed it in common as that dir was housing things that the canvas_site_creator and bulk_site_creator apps shared but it probably is better placed at the project level. 
I moved the model into canvas_account_admin_tools/models.py and updated import references

Chris-Thornton-Harvard commented:
Note: since the canvas_school_template table already exists, a fake migrate should be run. This will insert a record into the django_migrations table for the model but not apply any changes
`python manage.py migrate  canvas_account_admin_tools --fake `
